src/main/java/io/cryostat/net/TargetConnectionManager.java

```diff
@@ -164,16 +176,80 @@ private JFRConnection attemptConnectAsHostPortPair(ConnectionDescriptor connecti
     private JFRConnection connect(
             ConnectionDescriptor cacheKey, JMXServiceURL url, Optional<Credentials> credentials)
             throws Exception {
+        JMXConnectionOpened evtOpened = new JMXConnectionOpened(url.toString());
+        JMXConnectionFailed evtFailed = new JMXConnectionFailed(url.toString());
+
         logger.info("Creating connection for {}", url.toString());
-        return jfrConnectionToolkit
-                .get()
-                .connect(
-                        url,
-                        credentials.orElse(null),
-                        Collections.singletonList(() -> this.connections.invalidate(cacheKey)));
+        evtOpened.begin();
+        evtFailed.begin();
+        try {
+            JFRConnection connection =
+                    jfrConnectionToolkit
+                            .get()
+                            .connect(
+                                    url,
+                                    credentials.orElse(null),
+                                    Collections.singletonList(
+                                            () -> this.connections.invalidate(cacheKey)));
+            evtOpened.end();
+            evtFailed.end();
+            if (evtOpened.shouldCommit()) {
+                evtOpened.commit();
+            }
+            return connection;
+        } catch (Exception e) {
+            evtOpened.end();
+            evtFailed.end();
+            if (evtFailed.shouldCommit()) {
+                evtFailed.commit();
+            }
+            throw e;
+        }
     }
 
     public interface ConnectedTask<T> {
         T execute(JFRConnection connection) throws Exception;
     }
+
+    @Name("io.cryostat.net.TargetConnectionManager.JMXConnectionOpened")
+    @Label("JMX Connection Status")
+    @Category("Cryostat")
+    @SuppressFBWarnings(
+            value = "URF_UNREAD_FIELD",
+            justification = "The event fields are recorded with JFR instead of accessed directly")
+    public static class JMXConnectionOpened extends Event {
+        String JMXServiceURL;
```

Comment:
Variable/field names like this should be named starting with lowercase letters, so in this case `jmxServiceUrl` or maybe `jmxServiceURL`.

However, since #521, I would also recommend changing this to `serviceUri` or something instead to be a bit more future-proof. Same goes for the same-named field in the Closed event.